🏗📖 Auto-format all markdown files in `ampproject/amphtml` with Prettier

rsimha opened the pull request:
The AMP Project already uses [Prettier](https://prettier.io/) to [auto-format](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#code-quality-and-style) JS files and several non-JS files across the repo. This PR adds documentation files to the mix. 

**Highlights:**
- Includes `.md` files in the list of files that are checked by Prettier
- Auto-formats all `.md` files in the repo by running `gulp prettier --fix`
- Skips the broken links check for PRs that modify a large number of `.md` files
- Prints suggested fixes when formatting errors are detected
- Exempts GitHub issue templates (they use a custom header that's incompatible with Prettier)
- Make extension generator (`gulp make-extension`) resilient to markdown file auto-fixing

**Notes:**
- Most of these changes do not affect the rendered HTML. Exceptions are:
    - Attribute lists are changed to all-in-one-line if they all fit on one line
![image](https://user-images.githubusercontent.com/26553114/67242799-42e7d380-f424-11e9-8596-7d001c5644a8.png)
    - Attribute lists are changed to one-per-line if they don't all fit on one line
![image](https://user-images.githubusercontent.com/26553114/67242852-601ca200-f424-11e9-93d6-1d73690ff7cf.png)
- Once this PR is merged, `.md` files will have to satisfy Prettier's formatting requirements
- A workflow to make your IDE auto-fix `.md` files on save is available [here](https://github.com/ampproject/amphtml/blob/master/contributing/getting-started-e2e.md#workflow-for-visual-studio-code)
- An alternative is to manually run `gulp prettify --local_changes --fix`

**Example of a failing check:**
<img width="1362" alt="" src="https://user-images.githubusercontent.com/26553114/67417518-f96bc580-f596-11e9-8567-504d1e3a4d30.png">


Fixes #25148

rsimha commented:
So many [broken links](https://travis-ci.org/ampproject/amphtml/jobs/601024915#L1476) in our documentation :frowning_face:

rsimha commented:
I sanity checked several of the changes in this PR, and found no visible changes to the rendered HTML. This is now ready for review.

rsimha commented:
Rebased and added a function to print suggested fixes for badly formatted `.md` files.

mrjoro commented on .github/ISSUE_TEMPLATE/bug-report.md:
```diff
@@ -4,7 +4,6 @@ about: Used to report bugs in AMP.
 title: ''
 labels: 'Type: Bug'
 assignees: ''
-
```
are we sure the issue templates don't need this?

mrjoro commented on .github/ISSUE_TEMPLATE/cherry_pick_template.md:
```diff
@@ -10,26 +10,29 @@ MUST: Replace *everything* in angle brackets in the title AND body of this issue
-| Issue | PR  | Production? | RC? | [Release issue](https://github.com/ampproject/amphtml/labels/Type%3A%20Release) |
-| :---: | :-: | :---------: | :-: | :-----------: |
-| #<_ISSUE_NUMBER_> | #<_PR_NUMBER_> | **<YES/NO>** | **<YES/NO>** | #<_RELEASE_ISSUE_> |
+|       Issue       |       PR       | Production?  |     RC?      | [Release issue](https://github.com/ampproject/amphtml/labels/Type%3A%20Release) |
```
What's going on with the spacing here?  I assumed it was trying to give the same space to each column, but the hyphens below don't look the same length.

mrjoro commented on 3p/README.md:
```diff
@@ -47,7 +47,7 @@ The following aren't hard requirements, but are performance optimizations we sho
 - When creatives animate they should not use more than 2ms time per frame to allow for other animations to have sufficient time for concurrent animations.
 
-The better an ad network does on the above requirements, the earlier ads from it will be loaded into AMP. In other words: *The slower the ad loads and the more it interferes with the page, the later AMP will load it.*
+The better an ad network does on the above requirements, the earlier ads from it will be loaded into AMP. In other words: _The slower the ad loads and the more it interferes with the page, the later AMP will load it._
```
nm; I see both are italics and both are valid, so standardizing on one makes sense

rsimha commented:
Rebased, reverted changes to issue template headers, and exempted them from future checks.

rsimha commented:
Some final notes:
- Self-reviewed every file in `contributing/`, and most files in `ads/`, `extensions/`, etc.
- Found and fixed a handful of existing bugs that were exposed by auto-formatting (tabs instead of spaces, bad formatting, etc.)
- Added `prettier-ignore` overrides where required
- Verified that the majority of `.md` files have no visible changes when rendered
- Tested VSCode auto-fix-on-save workflow for `.md` files
- Tested `gulp prettify` with `--fix`, `--files`, and `--local_changes`

Merging this now :crossed_fingers:
